### DISCUSSION TITLE
help for the program running.

### DISCUSSION THREAD

**Discussion by mmcme** [roll-w/CSL2-StationNaming#17] [Q&A]
hi, 
thanks for your mod code.
i have download it, and i compile the project in visual studio 2022. 
it build successful, and copied the "StationNaming.dll" into folder "mods" automatically.
however, after i start the game, i can not find the setting ui. 
and the program seems works.

before that  i new the file "StationNaming.coc" in the folder "C:\Users\Administrator\AppData\LocalLow\Colossal Order\Cities Skylines II"
the content is following:
StationNaming
{
    "Prefix": " ",
    "Suffix": " ",
    "NamingSeparator": "-",
    "NamedAddressNameFormat": "{NAME}, {NUMBER} {ROAD}",
    "AddressNameFormat": "{NUMBER} {ROAD}",
    "RoadFormat": {
        "Separator": " ",
        "Prefix": "",
        "Suffix": ""
    },
    "DistrictFormat": {
        "Separator": " ",
        "Prefix": "",
        "Suffix": ""
    }
}

so i don`t  know what happened, and how to fix it.
is there any other framework neccessary ?

**Comment by roll-w** [comment]
First, ensure that you have installed the Modding toolchain. The Setting UI should be visible directly in the game after build, without any additional steps. The in-game UI requires you to install npm before you can build it.

**Reply by mmcme** [reply]
hi
thanks for your reply.
when i built the project, i saw the "npm webpack" issue. however, i thought its for linux build. i just notice the file .dll is ok.

after your reply, i installed npm and webpack.  however, visual studio 2022 output:Cannot find module 'mini-css-extract-plugin'
then i installed mini-css-extract-plugin with npm. but the problem is still there.
could you offer more information?

following is the details of output while compiling:
Assembly post processed
1>
1>Start Burst compiler
1>
1>Start Burst compiler for Windows
1>Compiled 1 library, containing 0 methods, in 3228ms using 16 threads: G:\program code\city2\CSL2-StationNaming-master\bin\Debug\net472\Temp\StationNaming_win_x86_64.dll,G:\program code\city2\CSL2-StationNaming-master\bin\Debug\net472\Temp\StationNaming_win_x86_64.pdb
1>
1>Burst dll compiled for Windows
1>
1>Start Burst compiler for macOS
1>Compiled 1 library, containing 0 methods, in 2363ms using 16 threads: G:\program code\city2\CSL2-StationNaming-master\bin\Debug\net472\Temp\StationNaming_mac_x86_64.bundle
1>
1>Burst dll compiled for macOS
1>
1>Start Burst compiler for Linux
1>Compiled 1 library, containing 0 methods, in 2819ms using 16 threads: G:\program code\city2\CSL2-StationNaming-master\bin\Debug\net472\Temp\StationNaming_linux_x86_64.so
1>
1>Burst dll compiled for Linux
1>
1>StationNaming.dll processing complete
1>Copy output to deploy directory C:\Users\Administrator\AppData\LocalLow\Colossal Order\Cities Skylines II\Mods\StationNaming
1>
1>> build
1>> webpack
1>
1>[webpack-cli] [31mFailed to load 'G:\program code\city2\CSL2-StationNaming-master\UI\webpack.config.js' config [39m
1>[webpack-cli] [31mError: Cannot find module 'mini-css-extract-plugin'
1>Require stack:
1>- G:\program code\city2\CSL2-StationNaming-master\UI\webpack.config.js
1>- D:\Program Files\nodejs\node_global\node_modules\webpack-cli\lib\webpack-cli.js
1>- D:\Program Files\nodejs\node_global\node_modules\webpack-cli\lib\bootstrap.js
1>- D:\Program Files\nodejs\node_global\node_modules\webpack-cli\bin\cli.js
1>- D:\Program Files\nodejs\node_global\node_modules\webpack\bin\webpack.js
1>    at Module._resolveFilename (node:internal/modules/cjs/loader:1225:15)
1>    at Module._load (node:internal/modules/cjs/loader:1051:27)
1>    at Module.require (node:internal/modules/cjs/loader:1311:19)
1>    at require (node:internal/modules/helpers:179:18)
1>    at Object.<anonymous> (G:\program code\city2\CSL2-StationNaming-master\UI\webpack.config.js:3:30)
1>    at Module._compile (node:internal/modules/cjs/loader:1469:14)
1>    at Module._extensions..js (node:internal/modules/cjs/loader:1548:10)
1>    at Module.load (node:internal/modules/cjs/loader:1288:32)
1>    at Module._load (node:internal/modules/cjs/loader:1104:12)
1>    at Module.require (node:internal/modules/cjs/loader:1311:19) {
1>  code: 'MODULE_NOT_FOUND',
1>  requireStack: [
1>    'G:\\program code\\city2\\CSL2-StationNaming-master\\UI\\webpack.config.js',
1>    'D:\\Program Files\\nodejs\\node_global\\node_modules\\webpack-cli\\lib\\webpack-cli.js',
1>    'D:\\Program Files\\nodejs\\node_global\\node_modules\\webpack-cli\\lib\\bootstrap.js',
1>    'D:\\Program Files\\nodejs\\node_global\\node_modules\\webpack-cli\\bin\\cli.js',
1>    'D:\\Program Files\\nodejs\\node_global\\node_modules\\webpack\\bin\\webpack.js'
1>  ]
1>}[39m
1>G:\program code\city2\CSL2-StationNaming-master\StationNaming.csproj(36,9): error MSB3073:  command “npm run build” quit. code: 2

**Reply by roll-w** [reply]
You may try to run `npm install` in the `UI` directory, then try build again.

**Reply by mmcme** [reply]
hi
thanks for your reply.
i run the command "npx create-csii-ui-mod' in the folder of vs project. 
it build successfully after that. no failure.
and there are two more files in the output folder: StationNaming.mjs, StationNaming.mjs.LICENSE.txt.
however there is still no any setting ui while i run the game.

the more details:


>> build
1>> webpack
1>
1>asset [1m[32mStationNaming.mjs[39m[22m 6.56 KiB [1m[32m[emitted][39m[22m [javascript module] [1m[32m[minimized][39m[22m (name: StationNaming) 1 related asset
1>orphan modules 13 KiB [1m[33m[orphan][39m[22m 7 modules
1>runtime modules 396 bytes 2 modules
1>built modules 14.5 KiB [1m[33m[built][39m[22m
1>  cacheable modules 1.05 KiB
1>    [1m./node_modules/react/jsx-runtime.js[39m[22m 214 bytes [1m[33m[built][39m[22m [1m[33m[code generated][39m[22m
1>    [1m./node_modules/react/cjs/react-jsx-runtime.production.min.js[39m[22m 859 bytes [1m[33m[built][39m[22m [1m[33m[code generated][39m[22m
1>  [1m./src/index.tsx + 7 modules[39m[22m 13.4 KiB [1m[31m[not cacheable][39m[22m [1m[33m[built][39m[22m [1m[33m[code generated][39m[22m
1>  [1mexternal "React"[39m[22m 42 bytes [1m[33m[built][39m[22m [1m[33m[code generated][39m[22m
1>webpack 5.95.0 compiled [1m[32msuccessfully[39m[22m in 1614 ms
1>
1>馃敤 Built StationNaming
1>   [90mC:\Users\Administrator\AppData\LocalLow\Colossal Order\Cities Skylines II\Mods\StationNaming[0m

**Reply by roll-w** [reply]
Looks like it built successfully, but if the settings UI still isn't showing up, I recommend you read the official wiki: [Modding](https://cs2.paradoxwikis.com/Modding) and [Options UI](https://cs2.paradoxwikis.com/Options_UI), and follow the guidelines there to create a mod that only contains the settings UI for test.

**Reply by cgthunderstrike** [reply]
help me

**Reply by cgthunderstrike** [reply]
i dont know how to use any of the code i find here and i would really like to but i am clueless

**Reply by roll-w** [reply]
> i dont know how to use any of the code i find here and i would really like to but i am clueless

If you have any questions on building this mod, you may start another discussion and we will discuss there. But for your question here, I also recommend you to read the official wiki first to get help.

**Reply by cgthunderstrike** [reply]
ok thank you
